lib/ext/ts.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+var fs = require('fs');
+
+var typescript;
+try {
+	typescript = require('typescript');
+} catch (err) {
+	console.log('AVA error: `--extensions=ts` require TypeScript to be installed.');
+	throw err;
+}
+
+var typescriptOptions = {
+	module: typescript.ModuleKind.CommonJS,
+	target: typescript.ScriptTarget.ES6,
+	sourceMap: true,
+	inlineSources: true,
+	inlineSourceMap: true
+};
+
+/**
+ *
+ * Ts-Node version
+ *
+ */
+function tsnodeRegister() {
+	try {
+		require('ts-node').register(typescriptOptions);
+	} catch (err) {
+		console.log('AVA error: `--extensions=ts` require ts-node to be installed.');
+		throw err;
+	}
+}
+
+function tsnodeTranspile(code, fileName) {
+	try {
+		const tsService = require('ts-node').register(typescriptOptions);
+		const result = tsService().compile(code, fileName);
+		return result;
+	} catch (err) {
+		console.log('AVA error: `--extensions=ts` require ts-node to be installed.');
+		throw err;
+	}
+}
+
+/**
+ *
+ * Typescript version
+ *
+ */
+function typescriptRegister() {
+	require.extensions['.ts'] = function (module, fileName) {
+		var code = fs.readFileSync(fileName, 'utf8');
+		var result = typescript.transpileModule(code, {
+			fileName,
+			compilerOptions: typescriptOptions,
+			reportDiagnostics: true
+		});
+		module._compile(result.outputText, fileName);
+	};
+
+	require('source-map-support').install({
+		hookRequire: true
+	});
+}
```

Comment:
yes and no.

ts-node will do this if we use transpile function from ts-node. However, the current version of lib/ext/ts.js is not using the ts-node by default, it use the raw typescript module, so we have to do this by ourselves.

I leave the ts-node in lib/ext/ts.js is because at the beginning I'm not sure which one is better, so I implement both, but at last I decided to use pure typescript by default:

```js
module.exports = module.exports.default = {
  register: typescriptRegister,
  transpile: typescriptTranspile,
```

Do you think we should use ts-node, or I need to get rid of the additional ts-node code in lib/ext/ts.js?